core/src/test/java/kafka/test/api/ShareConsumerTest.java

```diff
@@ -1542,14 +1629,135 @@ public void testLsoMovementByRecordsDeletion(String persister) {
+        KafkaProducer<byte[], byte[]> producer = createProducer(new ByteArraySerializer(), new ByteArraySerializer());
+        // Producing a record.
+        producer.send(record);
+        ConsumerRecords<byte[], byte[]> records = shareConsumer.poll(Duration.ofMillis(5000));
+        // Since the value for share.auto.offset.reset has been altered to "earliest", the consumer should consume
+        // all messages present on the partition
+        assertEquals(1, records.count());
+        // Producing another record.
+        producer.send(record);
+        records = shareConsumer.poll(Duration.ofMillis(5000));
+        // The next records should also be consumed successfully
+        assertEquals(1, records.count());
+        shareConsumer.close();
+        producer.close();
+    }
+
+    @ParameterizedTest(name = "{displayName}.persister={0}")
+    @ValueSource(strings = {NO_OP_PERSISTER, DEFAULT_STATE_PERSISTER})
+    public void testShareAutoOffsetResetEarliestAfterLsoMovement(String persister) throws Exception {
+        KafkaShareConsumer<byte[], byte[]> shareConsumer = createShareConsumer(new ByteArrayDeserializer(), new ByteArrayDeserializer(), "group1");
+        shareConsumer.subscribe(Collections.singleton(tp.topic()));
+        // Changing the value of share.auto.offset.reset value to "earliest"
+        alterShareAutoOffsetReset("group1", "earliest");
+        ProducerRecord<byte[], byte[]> record = new ProducerRecord<>(tp.topic(), tp.partition(), null, "key".getBytes(), "value".getBytes());
+        KafkaProducer<byte[], byte[]> producer = createProducer(new ByteArraySerializer(), new ByteArraySerializer());
+        // We write 10 records to the topic, so they would be written from offsets 0-9 on the topic.
+        try {
+            for (int i = 0; i < 10; i++) {
+                producer.send(record).get();
+            }
+        } catch (Exception e) {
+            fail("Failed to send records: " + e);
+        }
+
+        // We delete records before offset 5, so the LSO should move to 5.
+        adminClient.deleteRecords(Collections.singletonMap(tp, RecordsToDelete.beforeOffset(5L)));
+
+        AtomicInteger totalMessagesConsumed = new AtomicInteger(0);
+        CompletableFuture<Integer> future = new CompletableFuture<>();
+        consumeMessages(totalMessagesConsumed, 5, "group1", 1, 10, true, future);
+        // The records returned belong to offsets 5-9.
+        assertEquals(5, totalMessagesConsumed.get());
+        assertEquals(5, future.get());
+
+        shareConsumer.close();
+        producer.close();
+    }
+
+    @ParameterizedTest(name = "{displayName}.persister={0}")
+    @ValueSource(strings = {NO_OP_PERSISTER, DEFAULT_STATE_PERSISTER})
+    public void testShareAutoOffsetResetMultipleGroupsWithDifferentValue(String persister) {
+        KafkaShareConsumer<byte[], byte[]> shareConsumer1 = createShareConsumer(new ByteArrayDeserializer(), new ByteArrayDeserializer(), "group1");
```

Comment:
How about `shareConsumerEarliest` and `shareConsumerLatest`? Much more descriptive than 1 or 2.